pkg/cli/admin/migrate/icsp/icsp.go

```diff
@@ -0,0 +1,197 @@
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+var (
+	internalMigrateICSPLong = templates.LongDesc(`
+	Update imagecontentsourcepolicy file(s) to imagedigestmirrorset file(s). If --dest-dir is unset, the imagedigestmirrorset file(s) that can be added to a cluster will be written to file(s) under the current directory.
+	`)
+
+	internalMigrateICSPExample = templates.Examples(`
+	# update the imagecontentsourcepolicy.yaml to new imagedigestmirrorset file under directory mydir
+	oc adm migrate icsp imagecontentsourcepolicy.yaml --dest-dir mydir
+`)
+)
+
+type MigrateICSPOptions struct {
+	genericclioptions.IOStreams
+
+	ICSPFiles []string
+	DestDir   string
+}
+
+func NewCmdMigrateICSP(f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewMigrateICSPOptions(streams)
+	cmd := &cobra.Command{
+		Use:     "icsp",
+		Short:   "Update imagecontentsourcepolicy file(s) to imagedigestmirrorset file(s).",
+		Long:    internalMigrateICSPLong,
+		Example: internalMigrateICSPExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+	cmd.Flags().StringVar(&o.DestDir, "dest-dir", o.DestDir, "Set a specific directory on the local machine to write imagedigestmirrorset file(s) to.")
+
+	return cmd
+}
+
+func NewMigrateICSPOptions(streams genericclioptions.IOStreams) *MigrateICSPOptions {
+	return &MigrateICSPOptions{
+		IOStreams: streams,
+	}
+}
+
+func (o *MigrateICSPOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	if len(args) < 1 {
+		return fmt.Errorf("convert expects at least one argument, path to an imagecontentsourcepolicy file")
+	}
+	o.ICSPFiles = args
+	return nil
+}
+
+func (o *MigrateICSPOptions) Run() error {
+	if o.DestDir == "" {
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		o.DestDir = cwd
+	}
+	if err := o.ensureDirectoryViable(); err != nil {
+		return err
+	}
+	// ensure destination path exists
+	if err := os.MkdirAll(o.DestDir, os.ModePerm); err != nil {
+		return err
+	}
+
+	var multiErr error
+	for _, file := range o.ICSPFiles {
+		if err := func() error {
+			icsp, name, err := readICSPsFromFile(file)
+			if err != nil {
+				return err
+			}
+			idmsYml, err := generateIDMS(icsp)
+			if err != nil {
+				return err
+			}
+
+			fname := filepath.Join(o.DestDir, fmt.Sprintf("imagedigestmirrorset_%s.%06d.yaml", name, rand.Int63()))
+			if err = os.WriteFile(fname, idmsYml, os.ModePerm); err != nil {
+				defer func() {
+					os.Remove(fname)
+				}()
+				return fmt.Errorf("error writing ImageDigestMirrorSet: %v", err)
+			}
+			fmt.Fprintf(o.Out, "wrote ImageDigestMirrorSet to %s\n", fname)
+			return nil
+		}(); err != nil {
+			multiErr = multierror.Append(multiErr, err)
```

Comment:
Don't use that, but rather use something like here: https://github.com/openshift/oc/blob/10f2c4952d8ba95938c1a82083664218dc02ee37/pkg/cli/rollout/retry.go#L245